docs/security.md

```diff
@@ -0,0 +1,194 @@
+invalid configuration is detected, the Collector MUST fail to start as a
+protective mechanism.
+
+> Note: Issue
+> [#886](https://github.com/open-telemetry/opentelemetry-collector/issues/886)
+> proposes adding a default configuration to the binary.
+
+The configuration drives the Collector's behavior and care should be taken to
+ensure the configuration only enables the minimum set of capabilities and as
+such exposes the minimum set of required ports. In addition, any incoming or
+outgoing communication SHOULD leverage TLS and authentication.
+
+The Collector keeps the configuration in memory, but where the configuration is
+loaded from at start time depends on the packaging used. For example, in
+Kubernetes secrets and configmaps CAN be leveraged. In comparison, the Docker
+image embeds the configuration in the container where is it not stored in an
+encrypted manner by default.
+
+The configuration MAY contain sensitive information including:
+
+- Authentication information such as API tokens
+- TLS certificates including private keys
+
+Sensitive information SHOULD be stored securely such as on an encrypted
+filesystem or secret store. Environment variables CAN be used to handle
+sensitive and non-sensitive data as the Collector MUST support environment
+variable expansion.
+
+> For more information on environment variable expansion, see
+> [this](https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables)
+> documentation.
+
+Component developers MUST get configuration information from the Collector's
+configuration file. Component developers SHOULD leverage [configuration helper
+functions](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config).
+
+More information about configuration is provided in the following sections.
+
+## Permissions
+
+The Collector supports running as a custom user and SHOULD NOT be run as a
+root/admin user. For the majority of use-cases, the Collector SHOULD NOT require
+privileged access to function. Some components MAY require privileged access
+and care should be taken before enabling these components. Collector components
+MAY require external permissions including network access or RBAC.
+
+Component developers SHOULD minimize privileged access requirements and MUST
+document what requires privileged access and why.
+
+More information about permissions is provided in the following sections.
+
+## Receivers and Exporters
+
+Receivers and Exporters can be either push or pull-based. In either case, the
+connection established SHOULD be over a secure and authenticated channel.
+Unused receivers and exporters SHOULD be disabled to minimize the attack vector
+of the Collector.
+
+Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
+settings via configuration parameters. If these settings are available, care
+should be taken before modifying the default values. Improperly setting these
```

Comment:
@jstickler would probably say: "care should be taken" by whom?

Reply:
@jpkrohling   I see that I've taught you well!  👍

Reply:
Good call -- will fix.